src/clean-core/alloc_array.hh

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <initializer_list>
+
+#include <clean-core/algorithms.hh>
+#include <clean-core/allocator.hh>
+#include <clean-core/assert.hh>
+#include <clean-core/detail/container_impl_util.hh>
+#include <clean-core/forward.hh>
+#include <clean-core/span.hh>
+
+namespace cc
+{
+// heap-allocated (runtime) fixed-size array
+// backed by a cc::allocator
+template <class T>
+struct alloc_array
+{
+    static_assert(sizeof(T) > 0, "cannot make alloc_array of incomplete object");
+
+    alloc_array(cc::allocator* allocator = cc::system_allocator) : _allocator(allocator) { CC_CONTRACT(allocator != nullptr); }
```

Comment:
explicit?

Reply:
Made it explicit, but added a `.destroy()` method as this would otherwise make invalidating the array inconvenient (`= {}` no longer works)